core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -814,12 +814,28 @@ private Function compileFilter(
         if (filter != null && !filter.isThreadSafe()) {
             assert filterExpr != null;
             ObjList<Function> workerFilters = new ObjList<>();
-            for (int i = 0; i < workerCount; i++) {
-                restoreWhereClause(filterExpr); // restore original filters in node query models
-                Function workerFilter = compileBooleanFilter(filterExpr, metadata, executionContext);
-                workerFilters.extendAndSet(i, workerFilter);
-                assert filter.getClass() == workerFilter.getClass();
+            boolean supportDeepClone = true;
+            try {
+                for (int i = 0; i < workerCount; i++) {
+                    workerFilters.extendAndSet(i, filter.deepClone());
+                }
+            } catch (UnsupportedOperationException e) {
+                LOG.info().$("Failed to deepClone filter function for parallel worker").I$();
```

Comment:
This shouldn't be an info level message as it'd scary the user. Debug level is more appropriate here. Same with the below changes.